src/CMakeLists.txt

```diff
@@ -172,43 +172,53 @@ message(STATUS "path_lib_rdkafka ${path_lib_rdkafka}")
 message(STATUS "path_include_graylog_logger ${path_include_graylog_logger}")
 message(STATUS "path_lib_graylog_logger ${path_lib_graylog_logger}")
 
-
-
-set(flatbuffers_generated_headers "")
-
-# Search the dev env standard locations, but it will also look in PATH
+#Generate flatbuffer header files
+# Search the dev env standard locations for flatc, but it will also look in PATH
 find_program(flatc flatc PATHS "$ENV{flatc}" "$ENV{HOME}/.tools" "/opt/local/flatbuffers")
 message(STATUS "flatc in ${flatc}")
 
-set(schemas_subdir "schemas")
-set(head_out_dir "${CMAKE_CURRENT_BINARY_DIR}/${schemas_subdir}")
-file(MAKE_DIRECTORY ${head_out_dir})
-file(GLOB_RECURSE flatbuffers_schemata2 RELATIVE "${path_include_streaming_data_types}/schemas" "${path_include_streaming_data_types}/schemas/*.fbs")
-
-foreach (f0 ${flatbuffers_schemata2})
-	string(REGEX REPLACE "\\.fbs$" "" s0 ${f0})
-	#message(STATUS "s0: ${s0}")
-	set(fbs "${s0}.fbs")
-	set(fbh "${s0}_generated.h")
-	#message(STATUS "fbs: ${fbs}")
-	#message(STATUS "fbh: ${fbh}")
-	add_custom_command(
-		OUTPUT "${head_out_dir}/${fbh}"
-		COMMAND ${flatc} --cpp --gen-mutable --gen-name-strings --scoped-enums "${path_include_streaming_data_types}/schemas/${fbs}"
-		DEPENDS "${path_include_streaming_data_types}/schemas/${fbs}"
-		WORKING_DIRECTORY "${head_out_dir}"
-		COMMENT "Process ${fbs} using ${flatc}"
-	)
-	list(APPEND flatbuffers_generated_headers "${head_out_dir}/${fbh}")
-endforeach()
-
-add_custom_target(flatbuffers_generate ALL DEPENDS ${flatbuffers_generated_headers})
-
-
+if (EXISTS ${path_include_streaming_data_types})
+    message(STATUS "path_include_streaming_data_types ${path_include_streaming_data_types}")
+    set(flatbuffers_generated_headers "")
+    set(schemas_subdir "schemas")
+    set(head_out_dir "${CMAKE_CURRENT_BINARY_DIR}/${schemas_subdir}")
+    file(MAKE_DIRECTORY ${head_out_dir})
+    file(GLOB_RECURSE flatbuffers_schemata2 RELATIVE "${streaming-data-types_SOURCE_DIR}/schemas/*.fbs" "${path_include_streaming_data_types}/schemas" "${path_include_streaming_data_types}/schemas/*.fbs")
+
+    foreach (f0 ${flatbuffers_schemata2})
+    	string(REGEX REPLACE "\\.fbs$" "" s0 ${f0})
+    	#message(STATUS "s0: ${s0}")
+    	set(fbs "${s0}.fbs")
+    	set(fbh "${s0}_generated.h")
+    	#message(STATUS "fbs: ${fbs}")
+    	#message(STATUS "fbh: ${fbh}")
+    	add_custom_command(
+    		OUTPUT "${head_out_dir}/${fbh}"
+    		COMMAND ${flatc} --cpp --gen-mutable --gen-name-strings --scoped-enums "${path_include_streaming_data_types}/schemas/${fbs}"
+    		DEPENDS "${path_include_streaming_data_types}/schemas/${fbs}"
+    		WORKING_DIRECTORY "${head_out_dir}"
+    		COMMENT "Process ${fbs} using ${flatc}"
+    	)
+    	list(APPEND flatbuffers_generated_headers "${head_out_dir}/${fbh}")
+    endforeach()
+
+    add_custom_target(flatbuffers_generate ALL DEPENDS ${flatbuffers_generated_headers})
+else()
+    message(STATUS "Downloading streaming data types.")
```

Comment:
Taking care of the updated schema files, and only the updated ones, was the job of the code from about line 183 that is now moved inside the if clause.
So what if we allow to download the streaming data types (opt-in) and then set $path_include_streaming_data_types so that the existing code (181 to 205) can take care of recompiling the updated schema files?